cpp/src/interop/arrow_data_structures.cpp

```diff
@@ -234,6 +234,9 @@ arrow_column::arrow_column(cudf::column&& input,
         std::move(*schema->children[0]), std::move(input), stream, mr);
     }()}
 {
+  auto tmp     = from_arrow_device_column(&container->schema, &container->owner, stream, mr);
+  view_columns = std::move(tmp.get_deleter().owned_mem_);
+  cached_view  = *tmp;
```

Comment:
Should these variables initialized in the initializer list instead of in the constructor body?

Reply:
Typically initializing the required member variables like this is discouraged. We can create a static function to evaluate these variables and call it in the initializer list. For example:
```
struct cached_cudf_cview {
  owned_columns_t view_columns;  ///< Cached view that manages ownership of non-view-only data.
  table_view cached_view;
};

arrow_column {
  ...
  cached_cudf_cview cached_cv;

  arrow_column::arrow_column() : cached_cv{create_cached_cv(...)}  { /* constructor body */ }

  static cached_cudf_cview create_cached_cv(...) { ... }
```

Reply:
I had the same thought, but the main issue with your proposal is that to fix this you need to change the member to be a struct that contains both of these, which makes accessing harder. We can certainly consider it though.